src/components/form/fields/identifierField.jsx

```diff
@@ -20,7 +20,7 @@ import { createField } from '../fieldCreator';
 */
 
 function specializeRawIdentifier(withPhoneNumber, inputValue, telCall = _ => undefined, emailCall = _ => undefined, otherCall = _ => undefined) {
-    if (withPhoneNumber && (/^\+?[0-9]+/.test(inputValue))) {
+    if (withPhoneNumber && (/^\+?[0-9]+$/.test(inputValue))) {
```

Comment:
Notice the `$` at the end. Match the whole string, so `123abc` does not match and is not considered a phone number.